reflectapi/src/codegen/openapi.rs

```diff
@@ -466,28 +466,19 @@ impl Converter {
 
                     "uuid::Uuid" | "char" | "std::string::String" => Type::String,
                     "std::marker::PhantomData" | "std::tuple::Tuple0" => Type::Null,
-                    "std::vec::Vec" | "std::array::Array" | "std::collections::HashSet" => {
-                        Type::Array {
-                            items: Box::new(self.convert_type_ref(
-                                schema,
-                                kind,
-                                ty_ref.arguments().next().unwrap(),
-                            )),
-                            unique_items: prim.name() == "std::collections::HashSet",
-                        }
-                    }
-                    // Treat these as transparent wrappers.
-                    "std::boxed::Box" | "std::cell::Cell" | "std::cell::RefCell"
-                    | "std::sync::Mutex" | "std::sync::RwLock" | "std::rc::Rc"
-                    | "std::rc::Weak" | "std::sync::Arc" | "std::sync::Weak" => {
-                        return self.convert_type_ref(
+                    "std::vec::Vec"
+                    | "std::array::Array"
+                    | "std::collections::HashSet"
+                    | "std::collections::BTreeSet" => Type::Array {
```

Comment:
I see 2 issues here:
1. as far as I remember HashSet has fallback type defined (which should be Vec), and so the codegen should use fallback type instead of the original type. Otherwise, codegen code would need to be changed every time when a new rust specific data structure is added. Codegen should allow to work with new types which it does not know about but have fallbacks.
2. BTreeSet does not have the fallback type defined?

Reply:
Ok, I will change it such that `BTree` family falls back to the `Hash`es. The reason `Hashset` doesn't fallback is because we can set the `unique` property on openapi arrays